### PR TITLE
Department Filtering Improvements

### PR DESCRIPTION
**Pull request by nihadness**
## Current behavior before PR
Filtering by Department Logic currently works with cookies which is set in Python files and handled in browser views by JS function and it is hard to follow/understand how everything works/is set. Since the logic underwent changes several times, the code is not very understandable nor well-written.   

## Desired behavior after PR is merged
I tried to make the code _better_ and more understandable in some files... Also tried to use Department Filtering in a mote efficient way in AR listing view (instead of filtering inside `is_item_allowed` function, AR's will be filtered by department while querying them).

#TODO: Requires some more improvements
--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

File: bika/lims/browser/analysisrequest/analysisrequests.py [hunk 1]
```diff
@@ -58,10 +58,16 @@ def __init__(self, context, request):
...
-            "path": {"query": "/", "depth": 2},
             "cancellation_state": "active",
         }
 
+        # Filter by Department
+        if self.context.bika_setup.getAllowDepartmentFiltering():
+            deps = self.request.get('filter_by_department_info', '')
+            dep_uids = deps.split(",")
+            dep_query = {"query": dep_uids, "operator": "or"}
+            self.contentFilter['getDepartmentUIDs'] = dep_query
```
Comment (xispa) [inline bika/lims/browser/analysisrequest/analysisrequests.py]: Do you mind to create a separate function somewhere else to deal with all this? Would be easier later to add changes without having to go through all code searching for this filter. I also think moving all this department filtering stuff to an add-on would be better. You could use an adapter here then to apply not only department filtering, but other additional filters we could eventually have in future without code changes in `senaite.core`

Comment (nihadness) [reply bika/lims/browser/analysisrequest/analysisrequests.py]: @xispa , I just tried to make it more generic, but seems it would take some time to add a filtering Add-on. And since that enhancement about Filtering AR's in querying level is kind of important, I would suggest to have at least that change immediately... If you still think we should just follow the way creating new Add-On for filtering, I will start to work on it soon.
